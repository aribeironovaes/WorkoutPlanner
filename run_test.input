(1km, 100), (1km, 5min/km)
-- EOT
(200m, 100)(45s, 100)
-- EOT
(2mi, 6:00min/mi, threshold)
-- EOT
(2mi, 80)
-- EOT
[(6min, 100), (2min, 80)], (10min, 70)
-- EOT
(2mi, 7:00min/mi)
-- EOT
(4s, 10mph), (5s, 6min/mi)
-- EOT
(45min, 3:44min/km)
-- EOT
(60min, 6:00min/mi)
-- EOT
(60min, 6:00 min/mi)
-- EOT