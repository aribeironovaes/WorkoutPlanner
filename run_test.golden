Input: 
(1km, 100), (1km, 5min/km)
Normalized: 
(1km, 100)
(1km, 74.6)
AST: 
ArrayInterval(
	SimpleInterval(1km, 100%, )
	SimpleInterval(1km, 5min/km, )
)
IF (Avg): 0.8635076320941238
TSS: 10.8
TSS2: 11.3
Distance (Miles): 1.2mi
Distance (Kilometers): 2km
Duration (Sec): 523.6936292054402
----------------------------
Input: 
(200m, 100)(45s, 100)
Normalized: 
(200m, 100)
(45sec, 100)
AST: 
ArrayInterval(
	SimpleInterval(200m, 100%, )
	SimpleInterval(45sec, 100%, )
)
IF (Avg): 1
TSS: 2.5
TSS2: 2.5
Distance (Miles): 0.2mi
Distance (Kilometers): 0.4km
Duration (Sec): 89.73872584108804
----------------------------
Input: 
(2mi, 6:00min/mi, threshold)
Normalized: 
(2mi, 100, threshold)
AST: 
ArrayInterval(
	SimpleInterval(2mi, 6:00min/mi, threshold)
)
IF (Avg): 1
TSS: 20
TSS2: 20
Distance (Miles): 2mi
Distance (Kilometers): 3.2km
Duration (Sec): 720
----------------------------
Input: 
(2mi, 80)
Normalized: 
(2mi, 80)
AST: 
ArrayInterval(
	SimpleInterval(2mi, 80%, )
)
IF (Avg): 0.8
TSS: 16
TSS2: 16
Distance (Miles): 2mi
Distance (Kilometers): 3.2km
Duration (Sec): 900
----------------------------
Input: 
[(6min, 100), (2min, 80)], (10min, 70)
Normalized: 
[(6min, 100), (2min, 80)]
(10min, 70)
AST: 
ArrayInterval(
	ArrayInterval(
		SimpleInterval(6min, 100%, )
		SimpleInterval(2min, 80%, )
	)
	SimpleInterval(10min, 70%, )
)
IF (Avg): 0.8225975119502044
TSS: 20.3
TSS2: 19.7
Distance (Miles): 2.4mi
Distance (Kilometers): 3.9km
Duration (Sec): 1080
----------------------------
Input: 
(2mi, 7:00min/mi)
Normalized: 
(2mi, 85.7)
AST: 
ArrayInterval(
	SimpleInterval(2mi, 7:00min/mi, )
)
IF (Avg): 0.8571428571428571
TSS: 17.1
TSS2: 18
Distance (Miles): 2mi
Distance (Kilometers): 3.2km
Duration (Sec): 840
----------------------------
Input: 
(4s, 10mph), (5s, 6min/mi)
Normalized: 
(4sec, 100)
(5sec, 100)
AST: 
ArrayInterval(
	SimpleInterval(4sec, 10mi/h, )
	SimpleInterval(5sec, 6:00min/mi, )
)
IF (Avg): 1
TSS: 0.3
TSS2: 0.3
Distance (Miles): 0mi
Distance (Kilometers): 0km
Duration (Sec): 9
----------------------------
Input: 
(45min, 3:44min/km)
Normalized: 
(45min, 99.9)
AST: 
ArrayInterval(
	SimpleInterval(45min, 3.7min/km, )
)
IF (Avg): 0.9986322732385723
TSS: 74.8
TSS2: 74.9
Distance (Miles): 7.5mi
Distance (Kilometers): 12.1km
Duration (Sec): 2700
----------------------------
Input: 
(60min, 6:00min/mi)
Normalized: 
(1hr, 100)
AST: 
ArrayInterval(
	SimpleInterval(1hr, 6:00min/mi, )
)
IF (Avg): 1
TSS: 100
TSS2: 100
Distance (Miles): 10mi
Distance (Kilometers): 16.1km
Duration (Sec): 3600
----------------------------
Input: 
(60min, 6:00 min/mi)
-- EOTNormalized: 
(1hr, 100)
AST: 
ArrayInterval(
	SimpleInterval(1hr, 6:00min/mi, )
)
IF (Avg): 1
TSS: 100
TSS2: 100
Distance (Miles): 10mi
Distance (Kilometers): 16.1km
Duration (Sec): 3600
----------------------------
